nx/lib/nx/defn/grad.ex

```diff
@@ -338,6 +333,8 @@ defmodule Nx.Defn.Grad do
   @verify_grad Application.compile_env(:nx, :verify_grad, false)
 
   defp update_grads(op, args, ans, g, _to_grad_ids, grads) do
+    args = revectorize_args(args, ans)
```

Comment:
I would prefer to not revectorized everything on every operation. Is there any chance we could do in `broadcast` only?

Reply:
`[unbroadcast(x, Nx.multiply(g, y), ans), unbroadcast(y, Nx.multiply(g, x), ans)]`

Lines like this one make it so that `g` is vectorized and `y` is unvectorized but has axes with the same name, so things break there.